models/classes/runner/QtiRunnerService.php

```diff
@@ -148,13 +148,14 @@ private function loadItemData($itemRef, $path)
      * @return QtiRunnerServiceContext
      * @throws \common_Exception
      */
-    public function getServiceContext($testDefinitionUri, $testCompilationUri, $testExecutionUri)
+    public function getServiceContext($testDefinitionUri, $testCompilationUri, $testExecutionUri, $userUri = null)
     {
         // create a service context based on the provided URI
         // initialize the test session and related objects
         $serviceContext = new QtiRunnerServiceContext($testDefinitionUri, $testCompilationUri, $testExecutionUri);
         $serviceContext->setServiceManager($this->getServiceManager());
         $serviceContext->setTestConfig($this->getTestConfig());
+        $serviceContext->setUserUri($userUri);
```

Comment:
Would be great to have condition:
```
if ($userUri) {
$serviceContext->setUserUri($userUri);
}
```

Reply:
also I don't like that from now we can change user for the context, maybe you can do it in the __construct, don't you?